File: version_Java/sample_9_security/src/main/java/com/example/sample_9_security/repository/CustomerRepository.java
package com.example.sample_9_security.repository;

import com.example.sample_9_security.domain.Customer;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;

import java.util.List;

public interface CustomerRepository extends JpaRepository<Customer, Integer> {
    @Query("SELECT x FROM Customer x ORDER BY x.firstName, x.lastName")
    List<Customer> findAllOrderByName();

    @Query("SELECT x FROM Customer x ORDER BY x.firstName, x.lastName")
    Page<Customer> findAllOrderByName(Pageable pageable);
}

File: version_Kotlin/sample_spirng_data_jpa/src/main/kotlin/com/example/sample_spring_data_jpa/repository/CustomerRepository.kt
package com.example.sample_spring_data_jpa.repository

import com.example.sample_spring_data_jpa.domain.Customer
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.stereotype.Repository

@Repository
interface CustomerRepository : JpaRepository<Customer, Int>{

    @Query("SELECT x FROM Customer x ORDER BY x.firstName, x.lastName")
    fun findAllOrderByName(): List<Customer>

    @Query("SELECT x FROM Customer x ORDER BY x.firstName, x.lastName")
    fun findAllOrderByName(pageable: Pageable): Page<Customer>
}